radixdlt/src/main/java/com/radixdlt/middleware2/network/MessageCentralSyncVerticesRPCNetwork.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.middleware2.network;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import com.radixdlt.consensus.GetVerticesResponse;
+import com.radixdlt.consensus.SyncVerticesRPCRx;
+import com.radixdlt.consensus.SyncVerticesRPCSender;
+import com.radixdlt.consensus.Vertex;
+import com.radixdlt.consensus.VertexStore.GetVerticesRequest;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.crypto.Hash;
+import com.radixdlt.network.addressbook.AddressBook;
+import com.radixdlt.network.addressbook.Peer;
+import com.radixdlt.network.messaging.MessageCentral;
+import com.radixdlt.network.messaging.MessageListener;
+import com.radixdlt.universe.Universe;
+import io.reactivex.rxjava3.core.Observable;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+
+/**
+ * Network interface for syncing vertices using the MessageCentral
+ */
+@Singleton
+public class MessageCentralSyncVerticesRPCNetwork implements SyncVerticesRPCSender, SyncVerticesRPCRx {
+
+	private final ECPublicKey selfPublicKey;
+	private final int magic;
+	private final AddressBook addressBook;
+	private final MessageCentral messageCentral;
+	private Cache<Hash, Object> opaqueCache = CacheBuilder.newBuilder()
+		.expireAfterWrite(30, TimeUnit.SECONDS)
```

Comment:
This really should be configurable.

Reply:
Gonna hold off on making this configurable until we settle on right approach for handling RPCs. Don't want to prematurely overdesign here.